### PR TITLE
Add food price sort to restaurant page

### PR DESCRIPTION
**Pull request by JoshuaRamc**
Added a toggler that changes the way in which foods are sorted on the restaurant page by price. Default sort is by low-to-high price.

### CODE REVIEW

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 1]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
+          labelId="food-sorter"
+          id="food Sorter"
+          value={sortDirection}
+          label="foodOrder"
+          onChange={handleSortChange}
+        >
+          <MenuItem value={1}>Low-To-High</MenuItem>
+          <MenuItem value={-1}>High-To-Low</MenuItem>
+        </Select>
+      </FormControl>
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:51]: This should be in its own component.

Using menu items feels too bloated. Can you think of a cleaner UI?

**Review by DLi7077 — CHANGES_REQUESTED** [review]

File: client/src/pages/Restaurant/RestaurantView/index.js (hunk 1 above)
Comment (DLi7077) [reply client/src/pages/Restaurant/RestaurantView/index.js:51]: What I mean is that this is taking too much space
image

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 2]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
     <div style={{ display: "flex", flexDirection: "column", gap: "0.5rem" }}>
       <Header restaurant={restaurant} />
       <RestaurantInfo restaurant={restaurant} />
-      <BudgetFilter foodsInBudget={foodsInBudget} budget={budget} setBudget={setBudget}/>
+      <BudgetFilter
+        foodsInBudget={foodsInBudget}
+        budget={budget}
+        setBudget={setBudget}
+      />
+      <FormControl sx={{ m: 1, maxWidth: 140, minWidth: 100 }}>
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:39]: why the min max width?

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 3]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
+        <InputLabel id="demo-simple-select-helper-label">Price Sort</InputLabel>
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:40]: why this id?

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 4]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
+        <Select
+          labelId="food-sorter"
+          id="food Sorter"
+          value={sortDirection}
+          label="foodOrder"
+          onChange={handleSortChange}
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:46]: if you really to use select, use the "standard" variant. See how I did our restaurant sorts

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 5]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
 import { useState } from "react";
+import InputLabel from "@mui/material/InputLabel";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import Select from "@mui/material/Select";
 
 export default function RestaurantView({ restaurant, foods }) {
   const [budget, setBudget] = useState("");
-  const foodsInBudget = foods.filter((food) => {
+  const [sortDirection, setSortDirection] = useState("");
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:13]: default this to ascending price

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 6]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
+        <InputLabel id="demo-simple-select-helper-label">Price Sort</InputLabel>
+        <Select
+          labelId="food-sorter"
+          id="food Sorter"
+          value={sortDirection}
+          label="foodOrder"
+          onChange={handleSortChange}
+        >
+          <MenuItem value={1}>Low-To-High</MenuItem>
+          <MenuItem value={-1}>High-To-Low</MenuItem>
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:49]: just label these "Ascending" and "Descending"

File: client/src/pages/Restaurant/RestaurantView/index.js (hunk 2 above)
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:39]: why the min & max width?

File: client/src/pages/Restaurant/RestaurantView/index.js (hunk 3 above)
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:40]: why this id?

File: client/src/pages/Restaurant/RestaurantView/index.js [hunk 7]
```diff
@@ -3,21 +3,52 @@ import FoodList from "../FoodList";
...
+      <BudgetFilter
+        foodsInBudget={foodsInBudget}
+        budget={budget}
+        setBudget={setBudget}
+      />
+      <FormControl sx={{ m: 1, maxWidth: 140, minWidth: 100 }}>
+        <InputLabel id="demo-simple-select-helper-label">Price Sort</InputLabel>
+        <Select
+          labelId="food-sorter"
+          id="food Sorter"
```
Comment (DLi7077) [inline client/src/pages/Restaurant/RestaurantView/index.js:43]: ids are good for testing. If you're gonna write tests, add these ids when you're making the tests instead

**Review by DLi7077 — CHANGES_REQUESTED** [review]

**Comment by JoshuaRamc** [comment]
Updated UI for the sort button: {changes on click from downwards to upwards arrow} 
image

**Comment by DLi7077** [comment]
looks aight

**Review by DLi7077 — APPROVED** [review]
good work

**Pull request merged**